teamscale-gradle-plugin/MIGRATION_35.md

```diff
@@ -0,0 +1,80 @@
+# Migration guide for the `com.teamscale` Gradle plugin version 34 to 35
+Previously, the plugin provided a way to specify the partitions either globally for a project or on a per task level.
+Additionally, a `teamscaleReportUpload` was provided automatically on the root project that would lazily pick up all the reports and upload them to Teamscale.
+
+Due to the restrictions that are imposed by the Gradle configuration cache, we had to adjust the API of the plugin to provide compatibility with it.
+
+
+## Usage of `Property` for all inputs
+We switched all extension and task properties from plain types e.g. `boolean` to `Property<Boolean>` etc.
+In newer Gradle versions this should be handled transparently.
+If you are still using an older version, you might need to replace `partition = "Something"` with `partition.set("Something")`.
+For more details, please refer to the official documentation: [Understanding Properties and Providers](https://docs.gradle.org/current/userguide/properties_providers.html#understanding_properties).
+
+## The `teamscaleReportUpload` task
+The type of the upload task was renamed from `TeamscaleUploadTask` to `TeamscaleUpload`.
+`teamscaleReportUpload` is no longer automatically added.
+Instead, you need to define your own upload task for each partition.
+
+```groovy
+import com.teamscale.TeamscaleUpload
+// ...
+
+tasks.register("teamscaleReportUpload", TeamscaleUpload) {
+    partition = "Unit Tests"
+    message = "Gradle upload" // Optional
+    // If you don't collect testwise coverage
+    from(tasks.test) // Will upload the JUnit reports produced by the Test task
+    from(tasks.jacocoTestReport) // Will upload the report produced by the JacocoReport task
+    
+    // or if you do collect testwise coverage
+    from(tasks.tiaTests) // Will upload the Testwise Coverage report produced by the TestImpacted task
+}
+```
+
+TODO aggregate at top-level
+
+## Specifying partitions top-level
+The `teamscale.report` extension action was removed.
+The partition is now specified via the `TeamscaleUpload` task.
+
+When you use test impact analysis, you need to additionally replace `teamscale.report.testwiseCoverage`
+```groovy
+teamscale {
+    report {
+        testwiseCoverage {
+            partition = "Unit Tests"
+        }
+    }
+}
+```
+with one of the following:
+If you have multiple `TestImpacted` tasks that should receive the same partition:
+```groovy
+import com.teamscale.TestImpacted
+// ...
+tasks.withType(TestImpacted) {
+    partition = "Unit Tests"
+}
+```
+If there is only one `TestImpacted` task to which the partition should apply:
```

Comment:
Can't I just also do the above in that case? That would simplify the instructions.

Additionally: why do I need to still specify the partition in the TestImpacted task? Shouldn't it be enough to specify it in the upload task? If not, then it's not clear from the text when to specify the partition in which location and why.

Reply:
Also, why can I only do this for the TestImpacted task but not for JUnit results or Jacoco coverage? seems inconsintent

Reply:
Because for the TestImpacted task we also need the partition to retrieve impacted tests from Teamscale. That's why the JUnit task for example does not need this.